packages/trackerless-network/protos/NetworkRpc.proto

```diff
@@ -47,48 +43,44 @@ enum StreamMessageType {
   GROUP_KEY_RESPONSE = 2;
 }
 
-enum EncryptionType {
-  NONE = 0;
-  RSA = 1;
-  AES = 2;
+enum ContentType {
+  JSON = 0;
+  BINARY = 1;
 }
 
-message EncryptedGroupKey {
-  string groupKeyId = 1;
-  string encryptedGroupKeyHex = 2;
-  optional string serialized = 3;
+enum EncryptionType {
+  NONE = 0;
+  AES = 1;
 }
 
 message StreamMessage {
   StreamMessageType messageType = 1;
-  optional EncryptionType encryptionType = 2;
-  bytes content = 3;
-  string signature = 4;
-  MessageRef messageRef = 5;
-  optional MessageRef previousMessageRef = 6;
-  optional string groupKeyId = 7;
-  optional EncryptedGroupKey newGroupKey = 8;
+  ContentType contentType = 2;
+  EncryptionType encryptionType = 3;
+  bytes content = 4;
+  bytes signature = 5;
+  MessageRef messageRef = 6;
+  optional MessageRef previousMessageRef = 7;
+  optional string groupKeyId = 8;
+  optional EncryptedGroupKey newGroupKey = 9;
 }
 
 message GroupKeyRequest {
   string requestId = 1;
-  string recipient = 2;
-  string rsaPublicKey = 3;
+  bytes recipient = 2;
+  bytes rsaPublicKey = 3;
   repeated string groupKeyIds = 4;
 }
 
 message GroupKeyResponse {
   string requestId = 1;
-  string recipient = 2;
-  repeated EncryptedGroupKey encryptedGroupKeys = 3;
+  bytes recipient = 2;
+  repeated EncryptedGroupKey groupKeys = 3;
 }
 
-message Layer2Message {
-  Layer2Type type = 1;
-}
-
-enum Layer2Type {
-  Data = 0;
+message EncryptedGroupKey {
+  string groupKeyId = 1;
+  bytes data = 2;
 }
 
 message StreamHandshakeRequest {
```

Comment:
Rename all `randomGraphId` fields to `graphId`s? If there are no other graphs types? (And if there are other types, those other graphs would just have separate id space.)

Reply:
Also what kind of content is in `StreamHandshakeRequest#interleavingFrom`? That info could maybe part of the field name as it is not trivial. (the `interleaveTarget` fields in other entities are `PeerDescriptors`, but this field is not?)

Reply:
Only making changes related to StreamMessages in this PR.